.github/workflows/vhiveease_golang.yml

```diff
@@ -0,0 +1,42 @@
+name: Build and deploy vhive golang image
+
+on: 
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ main ]
+
+jobs:
+  build_images:
+    name: Build docker vhiveease/golang image
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+    steps:
+
+    - uses: actions/checkout@v2
+      with:
+        ref: main
+
+    - name: Login to Docker Hub
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USERNAME }}
+        password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v1
+
+    - name: Build and push
+      id: docker_build
+      uses: docker/build-push-action@v2
+      with:
+          context: ./configs/golang/
+          file: ./configs/golang/Dockerfile
+          push: true
+          tags: ${{ secrets.DOCKER_HUB_USERNAME }}/golang:latest
```

Comment:
please push another tag that represents the current version of the image in it's name. Make the name of the image an environment variable to be set on top of this file